pkg/networkserver/mac/utils.go

```diff
@@ -222,6 +222,32 @@ func DeviceDefaultPingSlotPeriodicity(dev *ttnpb.EndDevice, defaults ttnpb.MACSe
 	}
 }
 
+func DeviceUplinkDwellTime(dev *ttnpb.EndDevice, phy *band.Band, defaults ttnpb.MACSettings) *ttnpb.BoolValue {
+	switch {
+	case dev.GetMacSettings().GetUplinkDwellTime() != nil:
+		return &ttnpb.BoolValue{Value: dev.MacSettings.UplinkDwellTime.Value}
+	case defaults.UplinkDwellTime != nil:
+		return &ttnpb.BoolValue{Value: defaults.UplinkDwellTime.Value}
+	case phy.BootDwellTime.Uplinks != nil:
+		return &ttnpb.BoolValue{Value: *phy.BootDwellTime.Uplinks}
+	default:
+		return nil
+	}
+}
+
+func DeviceDownlinkDwellTime(dev *ttnpb.EndDevice, phy *band.Band, defaults ttnpb.MACSettings) *ttnpb.BoolValue {
+	switch {
+	case dev.GetMacSettings().GetDownlinkDwellTime() != nil:
+		return &ttnpb.BoolValue{Value: dev.MacSettings.DownlinkDwellTime.Value}
+	case defaults.DownlinkDwellTime != nil:
+		return &ttnpb.BoolValue{Value: defaults.DownlinkDwellTime.Value}
+	case phy.BootDwellTime.Downlinks != nil:
+		return &ttnpb.BoolValue{Value: *phy.BootDwellTime.Downlinks}
+	default:
+		return nil
+	}
+}
```

Comment:
Shouldn't we prioritize boot settings over default settings?

Do we need this part of default settings in the first place?

Reply:
Default settings are the Network Server default settings - in pretty much any parameter generation site we put them just bellow the device MAC settings, but above the band/frequency plan.

Keep in mind that this is for consistency - I didn't actually expose the setting in the NS config.